io.github.ladaapp.lada.yaml

```diff
@@ -0,0 +1,48 @@
+id: io.github.ladaapp.lada
+runtime: org.gnome.Platform
+runtime-version: '47'
+sdk: org.gnome.Sdk
+command: lada
+finish-args:
+  - "--socket=wayland"
+  - "--socket=fallback-x11"
+  - "--device=dri"
+  - "--share=ipc"
+  - "--socket=pulseaudio"
+modules:
+  - lada-pip-dependencies.json
+  - name: lada
+    buildsystem: simple
+    build-commands:
+      - |
+        mkdir -p /app/model_weights/3rd_party
+        mv lada_mosaic_detection_model.pt lada_mosaic_edge_detection_model.pth lada_mosaic_restoration_model_bj_pov.pth lada_mosaic_restoration_model_generic.pth /app/model_weights/
+        # mv clean_youknow_video.pth /app/model_weights/3rd_party/ # todo
+      - python3 -m pip install --prefix=/app --no-deps '.[basicvsrpp,gui]'
+      - |
+        install -d /app/share
+        mv flatpak/share/* /app/share/
+      - |
+        patch -i $FLATPAK_BUILDER_BUILDDIR/patches/adjust_mmengine_resume_dataloader.patch -p2 -d /app
+        patch -i $FLATPAK_BUILDER_BUILDDIR/patches/bump_mmagic_mmcv_dependency_bound.patch -p2 -d /app
+        patch -i $FLATPAK_BUILDER_BUILDDIR/patches/enable_loading_vgg19_from_local_file.patch -p2 -d /app
+        patch -i $FLATPAK_BUILDER_BUILDDIR/patches/fix_diffusers_import.patch -p2 -d /app
```

Comment:
use type: patch instead of manually doing this https://docs.flatpak.org/en/latest/module-sources.html#patch-sources

Make sure to have the correct path in the patch files.

Reply:
I've spent more than two hours trying to do this and couldn't make it run.
I cannot apply the patch directly on the build dir where the the files are downloaded into as these files are basically zip files.
Only after pip install the files are extracted to installed to the /app dir
I hoped to make it work by adding another module just to run these patches which runs after the above steps happened
```
  - name: patch-installed-pip-dependencies
    buildsystem: simple
    build-commands: []
    sources:
      - type: patch
        paths:
          - "../patches/adjust_mmengine_resume_dataloader.patch"
          - "../patches/bump_mmagic_mmcv_dependency_bound.patch"
          - "../patches/enable_loading_vgg19_from_local_file.patch"
          - "../patches/fix_diffusers_import.patch"
        strip-components: 2
        options: ["-d", "/app"]
```
Which I hoped to be the equivalent to my shell command version above
This didn't work though:
```
Applying patch adjust_mmengine_resume_dataloader.patch
can't find file to patch at input line 3
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|--- old/.venv/lib/python3.12/site-packages/mmengine/runner/loops.py	2024-09-22 22:11:32.120340983 +0200
|+++ new/.venv/lib/python3.12/site-packages/mmengine/runner/loops.py	2024-09-26 21:51:19.303710074 +0200
```

Reply:
why do you have patches in the git repository? 

It'd be easier to apply them in a different branches in your git repo and use the commit of that branch, in the `sources` here.

Reply:
Its easier to have the patches in my upstream repo for folks that don't use flatpak.
But there doesn't seem to be an issue in finding the patches only when they're applied.

Is there an issue with running patch in the `build-commands` instead of the git source?